### DISCUSSION TITLE
v2.0 and Platformio

### DISCUSSION THREAD

**Discussion by owennewo** [stm32duino/Arduino_Core_STM32#1363] [Q&A]
Congratulations for getting 2.0 over the line!   🎉🥳👏
A lot of hard work, I'm sure.

Am I right in thinking there will need to be a corresponding update to ststm32 before I can use it in platformio?

Also I helped add the b-g431-esc1 board.  Whilst the variant and periphmaps made it into 2.0, I think its missing a board entry, right?
https://github.com/stm32duino/Arduino_Core_STM32/blob/master/variants/STM32G4xx/G431C(6-8-B)U_G441CBU/boards_entry.txt

**Comment by fpistm** [comment]
> Congratulations for getting 2.0 over the line! 🎉🥳👏
> A lot of hard work, I'm sure.

Thanks. Yes too much. :wink:



> Am I right in thinking there will need to be a corresponding update to ststm32 before I can use it in platformio?

Yes. it is required. I guess @valeros will do it or maybe you should open an issue to request it. Anyway, there is a lot of change and I guess it will requires more time than before.

> Also I helped add the b-g431-esc1 board. Whilst the variant and periphmaps made it into 2.0, I think its missing a board entry, right?
> https://github.com/stm32duino/Arduino_Core_STM32/blob/master/variants/STM32G4xx/G431C(6-8-B)U_G441CBU/boards_entry.txt

No it is ok:
https://github.com/stm32duino/Arduino_Core_STM32/blob/2e9eeabc6c540e6157f83ec0ec08e29af47086e7/boards.txt#L731-L743

boards_entry.txt in the variant folder is a helper to declare generic one and is automatically generated. The boards.txt is the only one used by Arduino to declare it.

**Reply by owennewo** [reply]
I'm perhaps being dumb, but i couldn't find/select the board in Arduino ide.  Just the generic ones were showing

**Reply by ABOSTM** [reply]
Hi @owennewo ,
this is a **Discovery** board:
![image](https://user-images.githubusercontent.com/50730894/115872170-caf7d080-a441-11eb-98f9-9e0bb7c67500.png)

**Comment by owennewo** [comment]
Thanks, you've answered my question!
